codegen/masm/tests/test_basic_arithmetic.rs

```diff
@@ -0,0 +1,212 @@
+use air_codegen_masm::{code_gen, constants};
+use assembly::Assembler;
+use ir::AirIR;
+use parser::parse;
+use processor::{AdviceInputs, Kernel, MemAdviceProvider, Process, StackInputs};
+
+static ARITH_AIR: &str = "
+def Add
+
+trace_columns:
+    main: [a, b]
+
+public_inputs:
+    stack_inputs: [16]
+
+boundary_constraints:
+    enf a.first = 0
+
+integrity_constraints:
+    enf a + a = 0
+    enf a - a = 0
+    enf a * a = 0
+    enf a^3 = 0
+
+    enf b + a = 0
+    enf b - a = 0
+    enf b * a = 0
+";
+
+#[test]
+fn test_simple_arithmetic() {
+    let ast = parse(ARITH_AIR).expect("parse failed");
+    let ir = AirIR::new(ast).expect("build AirIR failed");
+    let code = code_gen(&ir).expect("codegen failed");
+
+    let (a, b) = (3u64, 7u64);
+    let code = format!(
+        "
+        begin
+            push.{}          # frame address
+            push.0.0.{b}.{a} # data
+            dup.4 mem_storew # save data to memory
+            dropw drop       # clean stack
+
+            # [generated code]
+            {}
+            # [/generated code]
+        end
+        ",
+        constants::OOD_FRAME_ADDRESS,
+        code,
+    );
+    let program = Assembler::default().compile(code).unwrap();
+
+    let mut process: Process<MemAdviceProvider> = Process::new(
+        Kernel::new(&[]),
+        StackInputs::new(vec![]),
+        AdviceInputs::default().into(),
+    );
+    let program_outputs = process.execute(&program).expect("execution failed");
+    let result_stack = program_outputs.stack();
+
+    // results are in stack-order
+    #[rustfmt::skip]
+    let expected = [
```

Comment:
The next thing to do is to load the delta from memory and multiply these elements to produce the numerator. Then add support for the aux (current code is assuming a Felt, supports for ext2 should be straight forward to add)